go-eicio/reader.go

```diff
@@ -3,46 +3,59 @@ package eicio
 import (
 	"encoding/binary"
 	"io"
+	"os"
 )
 
 type Reader struct {
 	byteReader io.Reader
 }
 
+func Open(filename string) (*Reader, error) {
+	file, err := os.Open(filename)
+	if err != nil {
+		return nil, err
+	}
+	return NewReader(file), nil
+}
+
+func (rdr *Reader) Close() {
+	rdr.byteReader.(*os.File).Close()
+}
+
 func NewReader(byteReader io.Reader) *Reader {
 	return &Reader{
 		byteReader: byteReader,
 	}
 }
 
-func (rdr *Reader) GetEvent() (event *Event, err error) {
+func (rdr *Reader) GetEvent() *Event {
```

Comment:
with this API change, it is a bit harder to know *why* the reader didn't manage to retrieve the `Event`.

perhaps you should add an `err error` field to the `Reader` and set it there?
(so at least one can get at the first error that occurred)